### PR TITLE
Add common case for knative authorization issue

### PR DESCRIPTION
**Pull request by fvde**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:
```release-note
Add section about common authorization issue related to knative and AWS ECR
```

### CODE REVIEW

**Comment by fvde** [comment]
@googlebot I signed it!

File: docs/KFSERVING_DEBUG_GUIDE.md [hunk 1]
```diff
@@ -108,6 +108,30 @@ Traceback (most recent call last):
...
 ```
 
+### Inference Service cannot fetch docker images from AWS ECR
+If you don't see the inference service created at all for custom images from private registries (such as AWS ECR), it might be that the Knative Serving Controller fails to authenticate itself against the registry.
+
+```bash
+failed to resolve image to digest: failed to fetch image information: unsupported status code 401; body: Not Authorized
+```
+
+You can verify that this is actually the case by spinning up a pod that uses your image. The pod should be able to fetch it, if the correct IAM roles are attached, while Knative is not able to. To circumvent this issue you can skip tag resolution.
```
Comment (yuzisun) [inline docs/KFSERVING_DEBUG_GUIDE.md:118]: @fvde Please help link this [knative doc](https://knative.dev/docs/serving/tag-resolution/) so people know what tag resolution means and what's the implication if you turn that off, thanks!

Comment (fvde) [reply docs/KFSERVING_DEBUG_GUIDE.md]: Done, thanks for the feedback! 👍

**Comment by yuzisun** [comment]
thanks @fvde for the doc contribution! This is very helpful.

/lgtm
/approve

**Pull request merged**